pages/instructorQuestionManualGrading/instructorQuestionManualGradingNextInstanceQuestion.sql

```diff
@@ -0,0 +1,14 @@
+-- BLOCK get_next_unmarked_instance_question
+SELECT iq.*
+    FROM instance_questions AS iq
+    JOIN variants AS v ON (v.instance_question_id = iq.id)
+    JOIN (
+        -- We only want the last submission...
+        SELECT * FROM submissions AS s
+        ORDER BY s.date DESC, s.id DESC
+        LIMIT 1
+    ) s ON (s.variant_id = v.id)
+    -- ... but ONLY IF the last submission 'graded_at' is null (has not been graded)
+WHERE iq.assessment_question_id = $assessment_question_id AND s.graded_at IS NULL
```

Comment:
probably wise to also include a filter for assessment_id, since that's what's being authorized here (otherwise I could use a course instance and assessment ID I have access to to bypass authz, and use the assessment question I want).

Reply:
Also (again, code style), have each condition on a separate line.

Reply:
This is quite smart and I like this idea. Done.